cmd/lifecycle/main.go

```diff
@@ -76,6 +82,90 @@ func subcommand(platform Platform) {
 	}
 }
 
+// handlers
+
+type DefaultCacheHandler struct {
+	keychain authn.Keychain
+}
+
+func NewCacheHandler(keychain authn.Keychain) *DefaultCacheHandler {
+	return &DefaultCacheHandler{
+		keychain: keychain,
+	}
+}
+
+func (ch *DefaultCacheHandler) InitCache(cacheImageRef string, cacheDir string) (lifecycle.Cache, error) {
+	var (
+		cacheStore lifecycle.Cache
+		err        error
+	)
+	if cacheImageRef != "" {
+		cacheStore, err = cache.NewImageCacheFromName(cacheImageRef, ch.keychain)
+		if err != nil {
+			return nil, errors.Wrap(err, "creating image cache")
+		}
+	} else if cacheDir != "" {
+		cacheStore, err = cache.NewVolumeCache(cacheDir)
+		if err != nil {
+			return nil, errors.Wrap(err, "creating volume cache")
+		}
+	}
+	return cacheStore, nil
+}
+
+type DefaultImageHandler struct {
+	docker   client.CommonAPIClient
+	keychain authn.Keychain
+}
+
+func NewImageHandler(docker client.CommonAPIClient, keychain authn.Keychain) *DefaultImageHandler {
+	return &DefaultImageHandler{
+		docker:   docker,
+		keychain: keychain,
+	}
+}
+
+func (h *DefaultImageHandler) InitImage(imageRef string) (imgutil.Image, error) {
+	if imageRef == "" {
+		return nil, nil
+	}
+
+	if h.docker != nil {
+		return local.NewImage(
+			imageRef,
+			h.docker,
+			local.FromBaseImage(imageRef),
+		)
+	}
+
+	return remote.NewImage(
+		imageRef,
+		h.keychain,
+		remote.FromBaseImage(imageRef),
+	)
+}
+
+func (h *DefaultImageHandler) Docker() bool {
+	return h.docker != nil
+}
+
+type DefaultConfigHandler struct{}
+
+func NewConfigHandler() *DefaultConfigHandler {
+	return &DefaultConfigHandler{}
+}
+
+func (h *DefaultConfigHandler) ReadGroup(path string) ([]buildpack.GroupBuildpack, error) {
```

Comment:
I wanted this to be in the lifecycle package, but its dependence on `verifyBuildpackApis` (and needing to return the correct error code) made things a bit complicated.